tools: add a glossary for dm (#1723)
pingcap/docs: dev/reference/tools/data-migration/glossary.md

---
title: TiDB Data Migration Glossary

> summary: Learn the list of special terms used in TiDB Data Migration.

Reviewer: ```suggestion
summary: Learn the terms used in TiDB Data Migration.
```

category: glossary
---

# TiDB Data Migration Glossary

> This page explains the special terms used in the logs, monitoring, configurations, and documentation of TiDB Data Migration (DM).

Reviewer: ```suggestion
This document lists the terms used in the logs, monitoring, configurations, and documentation of TiDB Data Migration (DM).
```

## B

### Binlog

> In TiDB DM, binlog refers to the binary log files generated in MySQL or MariaDB. Refer to [MySQL Binary Log](https://dev.mysql.com/doc/internals/en/binary-log.html) and [MariaDB Binary Log](https://mariadb.com/kb/en/library/binary-log/) for details.

Reviewer: ```suggestion
In TiDB DM, binlogs refer to the binary log files generated in the TiDB database. It has the same indications as MySQL or MariaDB does. Refer to [MySQL Binary Log](https://dev.mysql.com/doc/internals/en/binary-log.html) and [MariaDB Binary Log](https://mariadb.com/kb/en/library/binary-log/) for details.
```

### Binlog event

> Binlog events are information about data modification made to a MySQL or MariaDB server instance stored in the binlog files. Refer to [MySQL Binlog Event](https://dev.mysql.com/doc/internals/en/binlog-event.html) and [MariaDB Binlog Event](https://mariadb.com/kb/en/library/1-binlog-events/) for details.

Reviewer: ```suggestion
Binlog events are information about data modification made to a MySQL or MariaDB server instance. These binlog events are stored in the binlog files. Refer to [MySQL Binlog Event](https://dev.mysql.com/doc/internals/en/binlog-event.html) and [MariaDB Binlog Event](https://mariadb.com/kb/en/library/1-binlog-events/) for details.
```

### Binlog event filter

> [Binlog event filter](/dev/reference/tools/data-migration/features/overview.md#binlog-event-filter) is a more fine-grained filtering rule than the black and white lists filtering rule. Refer to [binlog event filter](/dev/reference/tools/data-migration/overview.md#binlog-event-filtering) for details.

Reviewer 2: Is `rule` accurate here? in Chinese, `比 Black & white table list 更加细粒度的过滤功能`.

### Binlog position

The binlog position is the offset information of a binlog event in a binlog file. Refer to [MySQL `SHOW BINLOG EVENTS`](https://dev.mysql.com/doc/refman/8.0/en/show-binlog-events.html) and [MariaDB `SHOW BINLOG EVENTS`](https://mariadb.com/kb/en/library/show-binlog-events/) for details.

### Binlog replication processing unit

> Binlog replication processing unit is the processing unit internally used in DM-worker to read upstream binlogs or local relay logs, and to migrate these logs to the downstream. Each subtask corresponds to a binlog replication processing unit. In the current documentation, the binlog replication processing unit is also referred to as the sync processing unit.

Reviewer 2: How about using `replicate` instead of `migrate`?

> Binlog replication processing unit is the processing unit internally used in DM-worker to read upstream binlogs or local relay logs, and to replicate these logs to the downstream. Each subtask corresponds to a binlog replication processing unit. In the current documentation, the binlog replication processing unit is also referred to as the sync processing unit.

Reviewer: ```suggestion
Binlog replication processing unit is the processing unit used in DM-worker to read upstream binlogs or local relay logs, and to replicate these logs to the downstream. Each subtask corresponds to a binlog replication processing unit. In the current documentation, the binlog replication processing unit is also referred to as the sync processing unit.
```
`in` already indicates the meaning of "internally".

### Black & white table list

Black & white table list is the feature that filters or only replicates all operations of some databases or some tables. Refer to [black & white table lists](/dev/reference/tools/data-migration/overview.md#black-and-white-lists-replication-at-the-schema-and-table-levels) for details. This feature is similar to [MySQL Replication Filtering](https://dev.mysql.com/doc/refman/5.6/en/replication-rules.html) and [MariaDB Replication Filters](https://mariadb.com/kb/en/library/replication-filters/).

## C

### Checkpoint

A checkpoint indicates the position from which a full data import or an incremental replication task is paused and resumed, or is stopped and restarted.

> - In a full import task, a checkpoint indicates the offset and other information of the successfully imported data in a file that is being imported. A checkpoint is updated in sync with the data import task.

Reviewer 2: How about using `synchronously` instead of `in sync`?

> - In an incremental replication, a checkpoint corresponds to the [binlog position](#binlog-position) and other information of a [binlog event](#binlog-event) that is successfully parsed and replicated to the downstream. A checkpoint is updated after the DDL operation is successfully imported or 30 seconds after the last update.

Reviewer 2: ```suggestion
- In an incremental replication, a checkpoint corresponds to the [binlog position](#binlog-position) and other information of a [binlog event](#binlog-event) that is successfully parsed and replicated to the downstream. A checkpoint is updated after the DDL operation is successfully replicated or 30 seconds after the last update.
```

> In addition, the `relay.meta` information corresponding to a [relay processing unit](#relay-processing-unit) works similarly to a checkpoint. A relay processing unit pulls the [binlog event](#binlog-event) from the upstream and writes this event to the [relay log](#relay-log), and writes the [binlog position](#binlog-position) corresponding to this event or the GTID information to `relay.meta`.

Reviewer 2: ```suggestion
In addition, the `relay.meta` information corresponding to a [relay processing unit](#relay-processing-unit) works similarly to a checkpoint. A relay processing unit pulls the [binlog event](#binlog-event) from the upstream and writes this event to the [relay log](#relay-log), and writes the [binlog position](#binlog-position) or the GTID information corresponding to this event to `relay.meta`.
```

## D

### Dump processing unit

> The dump processing unit is the processing unit internally used in DM-worker to migrate all data from the upstream. Each subtask corresponds to a dump processing unit.

Reviewer 2: any words to indicate `导出`?

The load processing unit is the processing unit internally used in DM-worker to import the fully exported data to the downstream. Each subtask corresponds to a load processing unit. In the current documentation, the load processing unit is also referred to as the import processing unit.

## R

### Relay log

The relay log is the binlog files that DM-worker pulls from the upstream MySQL or MariaDB, and stores in the local. The format of the relay log is the standard binlog file, which can be parsed by using the compatible [mysqlbinlog](https://dev.mysql.com/doc/refman/8.0/en/mysqlbinlog.html) and other tools.

Refer to [TiDB DM relay log](/dev/reference/tools/data-migration/relay-log.md) for more details about the directory structure, initial replication rules, and data purge.

### Relay processing unit

The relay processing unit is the processing unit internally used in DM-worker to pull binlogs from the upstream and write data into relay logs. Each DM-worker instance has only one relay processing unit.

## S

### Safe mode

Safe mode is the mode in which DML statements can be imported more than once when the primary key or unique key exists in the table schema.

> In this mode, some statements from the upstream are replicated to the downstream only after they are re-written. The `INSERT` statement is re-written as `REPLACE`; the `UPDATE` statement is re-written as `DELETE` and `REPLACE`. TiDB DM automatically enables the safe mode 5 minutes before the replication task is started or resumed. You can manually enable the mode by modifying the `safe-mode` parameter in the task configuration file.

Reviewer 2: Is `before` accurate here? It should be `the 5 minutes after the replication task is started or resumed`.
"在启动或恢复增量迁移任务 **成功后** 的前 5 分钟"

### Shard DDL

> The shard DDL is the DDL statements that are executed in the upstream and need to be coordinated and migrated by TiDB DM in the process of merging sharded tables. In the current documentation, the shard DDL is also referred to as the sharding DDL.

Reviewer: ```suggestion
The shard DDL is the DDL statement that is executed on the upstream sharded tables. It needs to be coordinated and migrated by TiDB DM in the process of merging the sharded tables. In the current documentation, the shard DDL is also referred to as the sharding DDL.
```

### Shard DDL lock

The shard DDL lock is the lock mechanism that coordinates the replication of shard DDL. Refer to [the implementation principles of merging and replicating data from sharded tables](/dev/reference/tools/data-migration/features/shard-merge.md#principles) for details. In the current documentation, the shard DDL lock is also referred to as the sharding DDL lock.

### Shard group

> A shard group is all the upstream sharded tables to be merged and replicated to the same table in the downstream. Two-level shard groups are used for implementation in TiDB DM. Refer to [the implementation principles of merging and replicating sharded tables](/dev/reference/tools/data-migration/features/shard-merge.md#principles) for details. In the current documentation, the shard group is also referred to as the sharding group.

Reviewer: ```suggestion
A shard group is all the upstream sharded tables to be merged and replicated to the same table in the downstream. Two-level shard groups are used for implementation of TiDB DM. Refer to [the implementation principles of merging and replicating sharded tables](/dev/reference/tools/data-migration/features/shard-merge.md#principles) for details. In the current documentation, the shard group is also referred to as the sharding group.
```

### Subtask

> The subtask is the part of a data replication task that is running on each DM-worker instance. In different task configurations, a single data replication task might have one subtask or multiple subtasks.

Reviewer: ```suggestion
The subtask is a part of a data replication task that is running on each DM-worker instance. In different task configurations, a single data replication task might have one subtask or multiple subtasks.
```

### Subtask status

> The subtask status is the status of a data replication subtask. The available status includes `New`, `Running`, `Paused`, `Stopped`, and `Finished`. Refer to [subtask status](/dev/reference/tools/data-migration/query-status.md#subtask-status) for more details about the status of a data replication task or subtask.

Reviewer: ```suggestion
The subtask status is the status of a data replication subtask. The current status options include `New`, `Running`, `Paused`, `Stopped`, and `Finished`. Refer to [subtask status](/dev/reference/tools/data-migration/query-status.md#subtask-status) for more details about the status of a data replication task or subtask.
```

## T

### Table routing

The table routing feature enables DM to replicate a certain table of the upstream MySQL or MariaDB instance to the specified table in the downstream, which can be used to merge and replicate sharded tables. Refer to [table routing](/dev/reference/tools/data-migration/features/overview.md#table-routing) for details.

### Task

The data replication task, which is started after you successfully execute a `start-task` command. In different task configurations, a single replication task can run on a single DM-worker instance or on multiple DM-worker instances at the same time.

### Task status

> The task status refers to the status of a data replication subtask. Refer to [subtask status](/dev/reference/tools/data-migration/query-status.md#subtask-status) for details.

Reviewer: ```suggestion
The task status refers to the status of a data replication task. The task status depends on the statuses of all its subtasks. Refer to [subtask status](/dev/reference/tools/data-migration/query-status.md#subtask-status) for details.
```
The original Chinese says "数据迁移子任务所处的状态". I think it should be "数据迁移任务所处的状态". Please check with the author on it. @TomShawn @csuzhangxc
